### ISSUE TITLE
Extracting data from planning documents

### ISSUE THREAD

**Issue by stevenjmesser** [digital-land/digital-land#360]
# **Summary**

MHCLG’s Digital Planning programme and DSIT’s [i.AI](https://ai.gov.uk/) are piloting new technical approaches to transforming planning information and maps from PDF and paper documents into data, using AI-powered digitalisation.

The aim of MHCLG’s Digital Planning programme is to make planning fit for the 21st century. One of its objectives is to have local planning authorities using modern planning software that can speed up the planning process, leading to more property development.

Software needs data to function. The Planning Data service ([https://planning.data.gov.uk](https://planning.data.gov.uk)) is working with 75 local planning authorities to make 4 priority datasets available to modern planning software, like PlanX. However, there is much more information which currently exists as planning documents in various formats: maps, paper, scans, microfiche, and PDF.

# **Problem statement**

The 311 local planning authorities across England may have recent [planning considerations](https://design.planning.data.gov.uk/planning-consideration/) and decisions available as data, but much of the historical information is held in documents – maps, paper, scans, microfiche, and PDF. This means it can’t be used in site assessment, development management or other digitally enabled planning processes. It requires a lot of manual effort to extract text and numbers from documents and turn them into datasets. 

The most time-consuming part is creating geospatial data from maps and drawings. In one exercise, the Planning Data team found it took [between 1 to 2 hours to manually draw the shapes for each conservation area](https://github.com/digital-land/conservation-area-data/blob/main/README.md#:~:text=Time%20consuming%20process) in a local planning authority area. The team also tried a more [automated approach using Python libraries](https://github.com/digital-land/conservation-area-data/blob/main/README.md#python-approach). After setting up a workflow and running it several times, the team was able to create more geospatial data – but it required much more quality control.

## Users this problem affects 

* Planning officers in local planning authorities need to spend thousands of hours converting documents into data  
* GIS officers and planning officers in local planning authorities need to spend thousands of hours checking and validating datasets  
* Planners, construction companies and developers need to spend time and money seeking historical planning data from local planning authorities

# **Opportunity** 

There is an opportunity to use generative AI to extract information from planning documents and add it to a dataset following a [data specification](https://digital-land.github.io/data-standards/about/what-is-a-data-standard#what-a-data-specification-wont-do). Frontier models, like Meta’s Segment Anything (SAM), have been developed to generate geospatial data out of maps and drawings. 

[Similar work in the US](https://dho.stanford.edu/wp-content/uploads/Covenants.pdf?) reduced digitising planning documents from $1.4 million and 10 years to just $258 dollars and 6 days.

If these approaches can generate data of sufficient quality, there is an opportunity to radically increase the amount of planning data available for new and existing digital products and services.

# **Incubator project** 

For the next 8-weeks, i.AI is [incubating](https://ai.gov.uk/about/#:~:text=AI%20solution%20hypotheses-,Incubation,-\(does%20the%20model) a project exploring the use of AI to automatically convert planning documents into modern Digital Planning data. They are approaching this by combining multiple state-of-the-art technologies into a single AI system.

At the core is Large Language Models (LLMs), specifically Vision Language Models (VLMs). These can adapt to many types of documents and accurately extract information, even from old historical documents saved as scanned images.

First, the LLMs are used to extract all text and date information whilst ensuring it adheres to Digital Planning’s data specifications.

Second, to adapt to the geospatial nature of planning documents, i.AI are combining an LLM with image analysis tools, for example, OpenCV and Segment Anything. These can extract boundary shapes (polygons) from maps within documents.

Next, to precisely georeference (geolocate) the boundary shapes to where they exist in the UK, i.AI are combining LLMs with tools such as OpenStreetMap and exploring novel techniques for map alignment and finding Ground Control Points automatically. 

By combining these frontier technologies in novel ways, i.AI can therefore extract text, dates, boundary shapes and locations from traditional documents. Initial prototypes show this can all be achieved in under 60 seconds, with plenty of room for further optimisation.

To ensure confidence in their results, i.AI are using an approach inspired by [test-driven development](https://gds-way.digital.cabinet-office.gov.uk/standards/test-driven-development.html) with a diverse [evaluation set](https://learn.microsoft.com/en-us/ai/playbook/technology-guidance/generative-ai/working-with-llms/evaluation/list-of-eval-metrics) of documents. This means they constantly assess performance across a wide variety of documents, whilst rapidly iterating to improve performance in a systematic way.

Repo: https://github.com/i-dot-ai/planning-extract
Project page: https://ai.gov.uk/projects/extract/

## Success criteria 

i.AI and MHCLG have set the following criteria to assess if they can digitise planning documents with AI.

### Accuracy criteria 

* Textual information is correct 95% of the time  
* Exact dates are correct at least 80% of the time  
* Dates are correct within one month at least 95% of the time   
* Polygon shapes achieve at least 0.8 Intersection over Union (IoU) on average  
* Georeferenced polygons are on average within 15m away  
* (Stretch) Georeferenced polygons are on average within 5m away

### Coverage criteria 

* The approach can accurately convert Article 4 Direction documents  
* The approach can accurately convert Conservation Area documents   
* (Stretch) The approach can adapt to Listed Building documents  
* (Stretch) The approach can adapt to Tree Preservation documents

## Guardrails 

The project is not involving any users yet, as this is purely exploring the potential for a technical solution, not a fully designed product or feature.

i.AI are building a representative evaluation set of planning documents from different local planning authorities, to capture the diversity of data across the country. They have also set up a robust evaluation pipeline to test both end-to-end performance and modular components of the system.

To mitigate the risk of copyright and licensing issues, the project is using planning documents and openly licensed datasets already provided by local planning authorities. Guidance has been provided by a data licensing expert on the Planning Data team. 

We’ll consider how to label data created using AI to comply with the [Planning Inspectorate’s guidance on use of artificial intelligence](https://www.gov.uk/guidance/use-of-artificial-intelligence-in-casework-evidence#:~:text=If%20you%20use%20AI%20to%20create%20or%20alter%20any%20part%20of%20your%20documents%2C%20information%20or%20data%2C%20you%20should%20tell%20us%20that%20you%20have%20done%20this%20when%20you%20provide%20the%20material%20to%20us.).

## What we’re not doing 

The aim of this project is to incubate a technical solution to prove or disprove the hypothesis that AI can be used to extract high-quality planning data from documents. As such, this project will not be looking at:

* the needs of users in local planning authorities  
* the needs of developers working on a data platform  
* copyright or licensing issues associated with the source material

We intend to look into these issues if the technical solution works.

If the project incubates a validated, working solution, we can take this forward to discover and design a usable, viable, valuable and feasible product or service.

**Comment by stevenjmesser** [comment]
Note from @GavEdwards:
> We're extending Extract's incubation stage for another 4 weeks 🌱 We have strong progress but haven't quite proven all the technology yet to claim the problem is fully solved. We have one key technical OKR left regarding 15m geolocation precision. Therefore, it's not quite ready to move to Alpha and user testing. However, we believe in the project and the technical and delivery progress up to now justifies another 4-week spike to prove (or disprove) this final technical challenge. The new end of incubation date will be 28 April.

**Comment by MrMikeRose** [comment]
I just want to flag here a couple of issues around intellectual property rights (IPR) and ownership here that need to be managed alongside the technical proof of this work. 

Currently, every single time this AI tool runs it creates a new dataset that is crown copyright... (as it is being done by the Crown) - and is probably 'administered' by DSIT (on behalf of the Crown) given this is where iAI sits. 

However this is compiled dataset from local authority data/documents that are owned by each of them - so potentially a compilation of data from circa 300 different owners. 
- Are we deleting that data? 
- Are we managing that data? 
- Do we have permission to create that data from owners? 
- What is the governance around this data?

Further, are we clear that this would be a tool run by local authorities rather than centrally (which would avoid ^) so the authorities would be in effect scraping data from their own documents. If so we need to be clear that any data created / compiled (ie that produced by the AI) in the process of testing the tool will be deleted (if it will).

Fundamentally, is there a decision making process / authority within MHCLG / DSIR for making / taking this kind of IPR decision - what I mean is - it might be an acceptable IPR risk to do this on behalf of LPAs and then data is created and shared back (still need to identify licensing approach or it would need to be an IPR transfer etc etc) but someone with the appropriate scheme of delegation within the departments should be documenting that decision and accepting that risk.

**Comment by stevenjmesser** [comment]
Recent conversations with local planning authorities have indicated that there may be 30–50 years' worth of planning information in documents, not available as data. This would indicate a need for the technical solution (if successful in this first stage) to handle a wider range of planning considerations than Article 4 Direction documents.

**Comment by stevenjmesser** [comment]
i.‌AI have published an explainer for the project on their website, [Extract](https://ai.gov.uk/projects/extract/).

**Comment by floehopper** [comment]
Probably not the right place for this, but it would be nice if the i.AI blog had an RSS feed! 🙏

**Comment by psd** [comment]
https://www.gov.uk/government/news/experimental-ai-could-help-councils-meet-housing-targets-by-digitising-records

**Comment by GavEdwards** [comment]
> Probably not the right place for this, but it would be nice if the i.AI blog had an RSS feed! 🙏

@floehopper This will be coming to our website very soon! Our team is running some tests before releasing the RSS feed

**Comment by stevenjmesser** [comment]
This [report on the survey 'Digitization and Artificial Intelligence for Archives
and Documentary Heritage Materials’](https://interparestrustai.org/assets/public/dissemination/RA03-InterPARESAI-Survey_Report_FINAL.pdf) could be useful in identifying risks and barriers associated with digitisation.

> All participants were asked to identify any barriers to using AI. Most respondents cited reasons linked to lack of resources or staffing issues, including lack of training or expertise (18.8%) and lack of staff time to integrate or use AI (18.3%). Other reasons, such as lack of funds (15.4%), lack of AI software and tools (15%) and lack of institutional awareness/support (14.5%) were also significant barriers.

**Comment by stevenjmesser** [comment]
Useful paper on [explainability methods for SAM](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=4878610).

**Comment by GavEdwards** [comment]
@floehopper RSS feed enabled ✅  https://ai.gov.uk/rss.xml

**Comment by stevenjmesser** [comment]
## Post-incubation report

The incubation period started with a original question: Can we create and precisely locate geospatial planning boundaries from documents using artificial intelligence?

After the incubation period ended on 29 April, we’re happy to say we’ve achieved that objective, based on our success criteria. i.‌AI have invented a novel method for automatically geolocating exactly where planning constraints apply. 

Using this method, AI can convert documents in a few minutes and it costs a few pence to run. As well as creating geospatial shapes, the workflow can also extract tabular data from documents against data specifications, such as [Article 4 Directions](https://digital-land.github.io/specification/specification/article-4-direction/).

## Results

### Evaluation pipeline

- Automated processing: 100% complete
- Modular evaluation system: 100% complete
- Article 4 Direction dataset: 100% complete
- Conservation Area dataset: 0%, paused whilst we decide on next steps

### Article 4 Direction conversion targets

- Text accuracy of 95%: 100% extraction of all expected text
- Date precision 80-95%: 94% of dates are extracted accurately
- Shape accuracy of 0.8 IoU: 90% of shapes are accurate
- Location within 10% of shape diameter (precise location): 82%, up 41%
- Location within 50% of shape diameter (general location): 94%, up 18%

### Document type coverage

- Conservation Areas: Paused
- Listed Buildings: Paused
- Tree Preservation Orders: Paused

## Next steps

Data is the foundational layer needed for a modern and efficient planning system. Yet this critical information, which informs what can and can’t be built, remains trapped in PDFs and paper documents. Converting each document manually takes 1–2 hours of valuable planning officer time — so, in practice, it simply isn’t happening for the vast majority of cases.

Below is an illustrative breakdown comparing human effort versus our AI workflow for digitising 100,000 documents:

| Metric | Human (1 planner) <br>8 h day · 5 d wk | AI (1 instance) <br>24 × 7 | Savings / Advantage |
|---|---|---|---|
| Processing time per doc | 2 h | 2 min | 1 h 58 min |
| Total labour / compute hours | 200 000 h | 3 333 h | 196 667 h<br>(-98.3 %) |
| Elapsed calendar time | ≈ 96.2 years | ≈ 139 days<br>(≈ 4.6 mo) | ↓ ≈ 95.8 years<br>(~ 252 × faster) |
| Cost per doc | £36 (£18/hour) | ~£0.10 | — |
| Total cost | £3 600 000 | £10 000 | £3 590 000 saved<br>(-99.72 %) |

_This breakdown includes assumptions which have not been validated. A proper evaluation will happen next and a validated benefits model can be produced._

The press release garnered a lot of interest from local planning authorities and helped us validate that this was a problem worth solving. 

A C-suite executive from one local authority told us that digital records of their statutory planning register date back to the early 2000s only. The rest is held in a basement. About 18 months ago they investigated a complete scanning exercise to get them digitised, but the cost was high (around £400k) alongside a significant commitment of planning officer time to review – which is better spent on planning, not digitising. 

Based on this, we’re taking Extract forward into an alpha phase. This will allow us to build and further evaluate it technically,  while running a short discovery into the unknowns we decided not to investigate during incubation (see ‘What we’re not doing’ at the start of this document).

**Comment by stevenjmesser** [comment]
## Alpha phase

The alpha phase is about testing the riskiest assumptions and trying out new approaches to solving a problem through prototyping. In the spirit of [treating discovery and alpha as a loop, not sequential stages](https://visitmy.website/2022/04/09/alpha-is-discovery/), we’ll also use the alpha phase to learn more about the problem space through exploring solutions.

> 🚧 This plan is in construction. We will update this page as the plan is shaped and refined with i.‌AI.

## Risky assumptions to test

We want to understand whether 

- Extract can help enough LPAs with digitisation and dataset creation to be cost-effective
- Extract can help LPAs turn a wide variety of documents into geospatial and tabular data
- Extract and its output can be trusted by anyone at an LPA who is involved in relevant or downstream processes (data, GIS, planning, IT, information knowledge management, legal, etc.)
- Extract and its output is trusted enough to use the data in real planning processes
- an AI literacy pack can help with forming and building trust
- LPAs need a human-in-the-loop quality check or a human-over-the-loop process would be satisfactory (based on the Local Land Charges model)
- LPAs could adopt an open-source technology and implement it themselves or they'd need it to be productised
- LPAs would need to hire other people or plan new processes in order to utilise it
- the copyright and IP risks are mitigated by offering it as a service
- the cost to run is huge or not

## Guardrails

We’ll conduct this phase in two stages: testing with one local authority and their staff first; if the first stage proves successful, we’ll invite two more authorities to join the alpha phase. 

Any and all evidence and needs collected will be useful in designing a product or service. However, if we learn that Extract is not valuable or viable, we can decide not to take it further. 

We will research perceived risks with LPAs and their relevant teams to understand what information needs to go into an [algorithmic transparency record](https://www.gov.uk/government/publications/guidance-for-organisations-using-the-algorithmic-transparency-recording-standard).

We will explore known copyright and intellectual property (IP) risks with an IP subject-matter expert on the team.

## Success criteria

### Technical OKRs

**State-of-the-Art (SOTA) extraction accuracy and speed**

- Textual information is correct 95% of the time
- Dates are the correct month and year 95% of the time
- Development rules are correct 95% of the time
- Georeferenced polygons are 0.8 IoU 90% of the time

**Extract reliably covers the four priority document types**
- Curated validation set of Article 4’s, Conservation Areas, TPOs &  Listed Buildings
- Achieve target accuracy on two of the validation datasets.

**Validate Extract’s real-world value**
- Survey LPAs to deeply understand the size of the problem. 
- A test set is built to evaluate Extract 
- Conduct and record user interviews with planners and GIS officers to shape Beta 

**Ship an open-source, MHCLG-ready product**
- The code is open source with an appropriate licence
- The code has high quality documentation
- A validation set is shared that is easily expandable by MHCLG
- An evaluation report is published
- A technical paper is published
- User requirements and feature sets required for Beta

**Comment by stevenjmesser** [comment]
Press release on GOV.‌UK: [PM unveils AI breakthrough to slash planning delays and help build 1.5 million homes](https://www.gov.uk/government/news/pm-unveils-ai-breakthrough-to-slash-planning-delays-and-help-build-15-million-homes-6-june-2025).

**Comment by stevenjmesser** [comment]
@GavEdwards detailed the R&D methodology of the incubation period in [our blog post](https://mhclgdigital.blog.gov.uk/2025/06/12/extract-using-ai-to-unlock-historic-planning-data/), sharing some of the methods around extracting textual data, image segmentation, and georeferencing.

**Comment by stevenjmesser** [comment]
We've created a home for our [design history and weeknotes](https://digital-land.github.io/extract-alpha/).

**Comment by stevenjmesser** [comment]
> I just want to flag here a couple of issues around intellectual property rights (IPR) and ownership here that need to be managed alongside the technical proof of this work.
> 
> Currently, every single time this AI tool runs it creates a new dataset that is crown copyright... (as it is being done by the Crown) - and is probably 'administered' by DSIT (on behalf of the Crown) given this is where iAI sits.
> 
> However this is compiled dataset from local authority data/documents that are owned by each of them - so potentially a compilation of data from circa 300 different owners.
> 
> * Are we deleting that data?
> * Are we managing that data?
> * Do we have permission to create that data from owners?
> * What is the governance around this data?
> 
> Further, are we clear that this would be a tool run by local authorities rather than centrally (which would avoid ^) so the authorities would be in effect scraping data from their own documents. If so we need to be clear that any data created / compiled (ie that produced by the AI) in the process of testing the tool will be deleted (if it will).
> 
> Fundamentally, is there a decision making process / authority within MHCLG / DSIR for making / taking this kind of IPR decision - what I mean is - it might be an acceptable IPR risk to do this on behalf of LPAs and then data is created and shared back (still need to identify licensing approach or it would need to be an IPR transfer etc etc) but someone with the appropriate scheme of delegation within the departments should be documenting that decision and accepting that risk.

Hi @MrMikeRose,

Grateful of your help in [reviewing the IP risks](https://github.com/digital-land/extract-alpha/issues/84). We can also look at the other issues you've highlighted.

Thanks in advance!
